server.js

```diff
@@ -5,17 +5,51 @@
  */
 
 const express = require('express')
+const request = require('request')
+const Config = require('./src/Config')
 const versoSpoof = require('./src/versoSpoof')
 
 const port = 8000
 const app = express()
 
-app.use(express.static(`${__dirname}/`))
+// Required for ElasticSearch proxy middleware to parse response body as JSON
+app.use(express.json())
+
+app.use('/api/search', (req, res) => {
+  if (!['GET', 'POST'].includes(req.method)) {
+    res.status(400).json({ error: `unsupported method: ${req.method}` })
+    return
+  }
+
+  // Hard-coded for now since we only have the one use case for proxying ES,
+  // i.e., full-text search of resources
+  if (req.path !== '/sinopia_resources/sinopia/_search') {
+    res.status(400).json({ error: `unsupported path: ${req.path}` })
+    return
+  }
 
-app.get('/api/search', (req, res) => {
-  res.json({ foo: 'bar' })
+  // Only use the method, path, and body from the original request: method and
+  // path have already been validated above and the body must be a
+  // JSON-serializeable entity
+  request({
+    method: req.method,
+    uri: `${Config.indexUrl}${req.path}`,
+    body: req.body,
+    json: true,
+  })
+    .on('error', (err) => {
+      console.error(`error making request to ElasticSearch: ${err}`)
+      res.status(500).json({ error: 'server error: could not make request to ElasticSearch' })
+    })
+    .pipe(res)
+    .on('error', (err) => {
+      console.error(`error returning ElasticSearch response: ${err}`)
+      res.status(500).json({ error: 'server error: could not send ElasticSearch response' })
+    })
 })
 
+app.use(express.static(`${__dirname}/`))
```

Comment:
I don't get what this line is doing

Reply:
@ndushay This middleware is for serving static assets to the browser, such as the stuff in `src/styles/` and `static/`. I'm not sure how "right" this code is given how we're using it, tbh, and I didn't add it—it was shifted around by my work. This code was added 4 years ago by LC: https://github.com/LD4P/sinopia_editor/commit/fb13bb2825df0e46e6e671d9258f4c957182cea0#diff-dc2704ac83baf8775944bab1251b45abR9

If we think we should tighten up how we serve static assets... new issue?

Reply:
how about just adding a comment:

```
//serve static assets to the browser, e.g. from src/styles/ and static/
```

And mayhaps making a ticket to move "styles" out of "src" directory??

Reply:
I was/am hoping someone with more knowledge of express could write up the ticket. I'm reluctant to pile that on the backlog without a clearer sense of the problem or the solution. (Sorry, still recovering from Hyrax backlog stress, where anyone and everyone tossed issues that were ill defined and often just questions.)

Reply:
ok.  I can live w/o the ticket, but would appreciate the comment.

Reply:
@ndushay ok.

Reply:
@ndushay :boom: https://github.com/LD4P/sinopia_editor/pull/887 :boom: